city_scrapers/spiders/fortx_Fort_Worth_Isd_Coc.py

```diff
@@ -0,0 +1,116 @@
+    ]
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+
+        location = {
+            "name": "Fort Worth ISD District Service Center",
+            "address": "7060 Camp Bowie Blvd, Fort Worth, TX 76116",
+        }
+
+        # go over the new events
+        for item in response.css(".fsDayContainer"):
+            meeting = Meeting(
+                title=self._parse_upcoming_title(item),
+                description="",
+                classification=COMMITTEE,
+                start=self._parse_upcoming_start(item),
+                end=self._parse_upcoming_end(item),
+                all_day=False,
+                time_notes="",
+                location=location,
+                links=[],
+                source=self._parse_source(response),
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+        # go over the old events
+        # the first table has the old events
+        for item in response.css("table")[0].css("tr"):
+            meeting = Meeting(
+                title="2021 COC",
+                description="",
+                classification=COMMITTEE,
+                start=self._parse_past_start(item),
+                end=None,
+                all_day=False,
+                time_notes="",
+                location=location,
+                links=self._parse_past_links(item),
+                source=self._parse_source(response),
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_upcoming_title(self, item):
+        """Parse or generate meeting title."""
+        return item.css(".fsTitle a::text").get()
+
+    def _parse_upcoming_start(self, item):
+        """Parse upcoming start datetime as a naive datetime object."""
+        datetime = item.css(".fsStartTime::attr(datetime)").get()
+        if not datetime:
+            return None
+        date, time = datetime.split("T")
+        time = time.split("-")[0]
+        return parse(f"{date} {time}")
+
+    def _parse_past_start(self, item):
+        """Parse past start date as a naive datetime object."""
+        date = item.css("td::text").get()
+        if not date:
+            return None
+        return parse(date)
+
+    def _parse_upcoming_end(self, item):
+        """Parse end datetime as a naive datetime object. Added by pipeline if None"""
+        datetime = item.css(".fsEndTime::attr(datetime)").get()
+        if not datetime:
+            return None
+        date, time = datetime.split("T")
+        time = time.split("-")[0]
+        return parse(f"{date} {time}")
+
+    def _parse_past_links(self, item):
```

Comment:
I recommend avoiding hardcoding the attachment names in this section. Instead, you can automatically retrieve this information using `cells[1].css("a::text").get()`.

Additionally, you can loop through the list of `td` elements in a table row and check whether each cell contains a `<a>` element. This way, you won't need to explicitly reference `cells[1]`, `cells[2]`, and `cells[3]`, and it also helps prevent an `IndexError` in case the number of returned cells is less than four.

Reply:
You're 100% right. Great advice. Will fix.